docker-java/src/test/java/com/github/dockerjava/cmd/swarm/CreateSecretCmdExecIT.java

```diff
@@ -49,4 +53,39 @@ public void testCreateSecret() throws Exception {
         assertThat(secretsAfterRemoved, IsCollectionWithSize.hasSize(0));
     }
 
+    @Test
+    public void testCreateSecretWithBinaryData() {
+        DockerClient dockerClient = startSwarm();
+
+        byte[] bytes = new byte[10];
+        new Random().nextBytes(bytes);
+
+        String secretName = UUID.randomUUID().toString();
+        CreateSecretResponse exec = dockerClient.createSecretCmd(
+            new SecretSpec()
+                .withName(secretName)
+                .withData(bytes))
+            .exec();
+
+        assertThat(exec, notNullValue());
+        assertThat(exec.getId(), notNullValue());
+
+        LOG.info("Secret created with ID {}", exec.getId());
+
+        List<Secret> secrets = dockerClient.listSecretsCmd()
+            .withNameFilter(Lists.newArrayList(secretName))
+            .exec();
+
+        assertThat(secrets, IsCollectionWithSize.hasSize(1));
+
+        dockerClient.removeSecretCmd(secrets.get(0).getId()).exec();
+
+        LOG.info("Secret removed with ID {}", exec.getId());
+
+        List<Secret> secretsAfterRemoved = dockerClient.listSecretsCmd()
+            .withNameFilter(Lists.newArrayList(secretName))
+            .exec();
+
+        assertThat(secretsAfterRemoved, IsCollectionWithSize.hasSize(0));
```

Comment:
there is no need to test list/delete here, but could you please verify that the content of the secret is equal to the original sequence of bytes?

Reply:
@bsideup testing the contents of the secret once it is is created is non-trivial - the contents is not available through the API (since it's secret) so instead the secret would need to be mounted into a container and output to the test via capture of STDOUT (which means monitoring the life-cycle of the container, retrieving the output, etc).

Do you feel all that is necessary or would it be sufficient to add a test for the SecretSpec itself? I don't believe there are any tests on the model currently, so this is a slightly new pattern of testing?

Reply:
@bsideup Apologies to chase for your input but it would be appreciated. As mentioned, there are no unit tests on the model currently. Extending the integration test to check the value of the secret would be non-trivial.